### PR TITLE
stop running random events when the shuttle leaves the station

### PR DESCRIPTION
**Pull request by BenLubar**
[BALANCE]

## About the PR

Once the shuttle leaves the station on its way back to CentCom, random events stop happening.

## Why's this needed?

Here is a very common scenario:

The round will end in 10 seconds.
Do you want to be a GHOST FROG? You have 60 seconds to decide.

or people becoming sleeper agents 10 seconds before the shuttle reaches CentCom.

## Changelog
```
(u)BenLubar:
(+)Random events no longer occur once the shuttle departs from the station or the round ends.
```

### CODE REVIEW

File: code/datums/controllers/event_controls.dm [hunk 1]
```diff
@@ -55,6 +55,11 @@ var/datum/event_controller/random_events
 			special_events += RE
 
 	proc/process()
+		if (emergency_shuttle.location > SHUTTLE_LOC_STATION)
```
Comment (ZeWaka) [inline code/datums/controllers/event_controls.dm:58]: this should also check for post-round game status i think

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/datums/controllers/event_controls.dm [hunk 2]
```diff
@@ -55,6 +55,11 @@ var/datum/event_controller/random_events
...
+			// you have 60 seconds to respond if you want to be a ghost frog
+			// also the server is rebooting in 10 seconds
```
Comment (mordent-goonstation) [inline code/datums/controllers/event_controls.dm:60]: I would count this comment as misleading.
```suggestion
			// prevent random events near round end
```

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

File: code/datums/controllers/event_controls.dm (hunk 1 above)
Comment (BenLubar) [reply code/datums/controllers/event_controls.dm:58]: Where's that status stored?

Comment (ZeWaka) [reply code/datums/controllers/event_controls.dm:58]: `current_state = GAME_STATE_FINISHED`

File: code/datums/controllers/event_controls.dm [hunk 3]
```diff
@@ -55,6 +55,10 @@ var/datum/event_controller/random_events
 			special_events += RE
 
 	proc/process()
+		if (emergency_shuttle.location > SHUTTLE_LOC_STATION || current_state == GAME_STATE_FINISHED)
+			// prevent random events near round end
```
Comment (ZeWaka) [inline code/datums/controllers/event_controls.dm:59]:
```suggestion
		// prevent random events near round end
		if (emergency_shuttle.location > SHUTTLE_LOC_STATION || current_state == GAME_STATE_FINISHED)
```

**Review by ZeWaka — APPROVED** [review]

**Review by mordent-goonstation — APPROVED** [review]

**Pull request merged**